doc/api/esm.md

```diff
@@ -689,6 +689,31 @@ stateless):
 }
 ```
 
+Another variant to also preserve backward-compatibility of `require`ing some
+files from the package that were previously available is to combine [Conditional
+Exports][] with [Nested conditions][] as follows:
+
+<!-- eslint-skip -->
+```js
+// This will allow to:
+// `import 'package'`
+// `require('package')`
+// `require('package/lib/file')`
+// while preventing to `import 'package/lib/file.js'`.
+{
+  "main": "./index.cjs",
+  "exports": {
+    ".": {
+      "import": "./wrapper.mjs",
+      "require": "./index.cjs"
+    },
+    "./lib": {
+      "require": "./lib"
```

Comment:
It makes sense, having API parity should probably be more important in this case. I understand the importance of promoting good patterns in our docs, so I guess we can close this? Or maybe there is a better place to put/adapt this example in the doc?

Reply:
There appears to be a good example of combining conditional exports and nested conditions.

https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_nested_conditions

One thing I liked about this example (and have thumbs upped it) is the naming scheme (`wrapper.mjs`, `index.cjs`).
It makes it very clear what is going on. I'm not sure the documentation for nested conditions is very clear in this respect. It may also be the case that the example is not necessarily using a wrapper module and this is why it is not explicitly named as such. Anyways, I like these kinds of PRs and hope to see more like them. :)

Reply:
👋 Sorry to bring this up again, but does anyone else in this PR thread think that it would be a good idea to replace the example shown in &hellip; 

https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_nested_conditions

&hellip; to one of the examples (or a slight derivative) of the one of the recommended dual-mode packages shown in &hellip;

https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_writing_dual_packages_while_avoiding_or_minimizing_hazards

&hellip; with the justification being that it further promotes well-established recommendations?

I've revised a couple of problem areas in the conditional exports section of `esm.md` already as a followup to my ambiguity concern in https://github.com/nodejs/node/pull/33812#discussion_r439768426 and could make such a change as a separate commit in that PR if it seems like a good idea.

Reply:
So, we've been having some more discussions and it is turning out to look like using the `"node"` condition while also setting a `"default"` condition may actually be a good best practice going forward after all. For more background see eg https://github.com/rollup/rollup/pull/3634#issuecomment-643811723.

We're still having these discussions though so I would wait just a little for things to settle. It is tentatively looking though like advising a nested `"node"` condition while always setting a `"default"` and then only providing `"browser"` when the build very specificlly is browser-specific will be the way to go.

Reply:
Note that there's a number of versions of node that either ignore, or warn on, the "node" condition :-/

Reply:
I don't think we should let experimental version compatibility affect forwards compatibility.

Reply:
Okay, so I won't be adding a commit for this in my PR. Let me know if you want me to open an issue `node/modules` about this.

> It is tentatively looking though like advising a nested "node" condition while always setting a "default" and then only providing "browser" when the build very specificlly is browser-specific will be the way to go.

+1

> I don't think we should let experimental version compatibility affect forwards compatibility.

There's a reason why the documentation is versioned. Things change. :)